redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -318,29 +318,41 @@ object Output {
       }
   }
 
-  final case class KeyValueTwoOutput[K, V](outK: Output[K], outV: Output[V]) extends Output[Map[K, V]] {
-    protected def tryDecode(respValue: RespValue)(implicit codec: Codec): Map[K, V] =
+  final case class StreamEntryOutput[I, K, V]()(implicit
+    idSchema: Schema[I],
+    keySchema: Schema[K],
+    valueSchema: Schema[V]
+  ) extends Output[StreamEntry[I, K, V]] {
+    protected def tryDecode(respValue: RespValue)(implicit codec: Codec): StreamEntry[I, K, V] =
       respValue match {
-        case RespValue.NullArray => Map.empty[K, V]
-        case RespValue.Array(entities) =>
-          entities.map {
-            case RespValue.Array(Seq(key @ RespValue.BulkString(_), value)) =>
-              outK.unsafeDecode(key) -> outV.unsafeDecode(value)
-            case other =>
-              throw ProtocolError(s"$other isn't a valid array")
-          }.toMap
-        case other => throw ProtocolError(s"$other isn't an array")
+        case RespValue.Array(Seq(id @ RespValue.BulkString(_), value)) =>
+          val eId   = ArbitraryOutput[I]().unsafeDecode(id)
+          val entry = KeyValueOutput(ArbitraryOutput[K](), ArbitraryOutput[V]()).unsafeDecode(value)
+          StreamEntry(id = eId, entry)
```

Comment:
```suggestion
          StreamEntry(eId, entry)
```
I'd also rename it to `entryId`.